# HG changeset patch
# User Todd C. Miller <Todd.Miller@sudo.ws>
# Date 1579208132 25200
# Node ID cc636a1af1b6fb9e5e89464b925ad6de66d37687
# Parent  c59e77060c375a2d0491ac9cb94a9063d5b2a1cc
Treat EROFS (like EACCES) as a non-fatal error in dir_is_writable().
Fixes sudoedit on macOS 10.15 and above where the root file system
is mounted read-only.  See https://support.apple.com/en-us/HT210650.
From Dan Villiom Podlaski Christiansen.  Bug #913

diff -r c59e77060c37 -r cc636a1af1b6 src/sudo_edit.c
--- src/sudo_edit.c	Wed Jan 15 14:47:42 2020 -0700
+++ src/sudo_edit.c	Thu Jan 16 13:55:32 2020 -0700
@@ -133,7 +133,7 @@
 
     if (rc == 0)
 	debug_return_int(true);
-    if (errno == EACCES)
+    if (errno == EACCES || errno == EROFS)
 	debug_return_int(false);
     debug_return_int(-1);
 }

